widget/playerctl.lua

```diff
@@ -71,17 +76,25 @@ local function factory(args)
                 
 
                 if state == "Playing" then
+
                     if notify == "on" and playerctl_now.title ~= helpers.get_map("current playerctl track") then
-                        helpers.set_map("current playerctl track", playerctl_now.title)
+                       local icon_cmd_full = string.format("%s%s", icon_cmd, playerctl_now.artUrl)
+
+			helpers.set_map("current playerctl track", playerctl_now.title)
+
+			helpers.async({ shell, "-c", icon_cmd_full }, function(cover_art)
+			end)
+
+			if followtag then playerctl_notification_preset.screen = focused() end
 
-                        if followtag then playerctl_notification_preset.screen = focused() end
-                        
-                        local common =  {
-                            preset      = playerctl_notification_preset,
-                            replaces_id = playerctl.id
-                        }
+			local common =  {
+				preset      = playerctl_notification_preset,
+				replaces_id = playerctl.id,
+				icon        = default_icon_file,
+				icon_size   = cover_size
+			}
 
-                        playerctl.id = naughty.notify(common).id
+			playerctl.id = naughty.notify(common).id
                     end
                 elseif state ~= "Paused" then
                     helpers.set_map("current playerctl track", nil)
```

Comment:
Need to delete default_icon_file on pause/stop, else if you skip to the next song sometimes it displays the last song's icon